core/shapes.cpp

```diff
@@ -428,12 +435,43 @@ void BoardShape::setupInserts() {
 
 const GCodeWriter BoardShape::generateBaseCutout(const Machine& machine) {
   auto tool = machine.tool(machine.baseCutoutTool());
-  auto paths = PathUtils::OffsetPath(buildOverallPath(machine), -0.2);
+  
+  // 1. Get path assuming full wrap of metal edge
+  auto fullWrapPath = PathUtils::OffsetPath(buildOverallPath(machine), -0.2);
+  assert(fullWrapPath.size() == 1);
+  
+  // 2. Form square outlining limit of metal edges (whole board if no partial edges)
+  MCFixed noseEdgeX = 0;
+  if (m_noseEdgeExt != 999.0) {
+    noseEdgeX += (m_noseLength - m_noseEdgeExt);
+  }
+  MCFixed tailEdgeX = m_noseLength + m_effectiveEdge + m_tailLength;
+  if (m_tailEdgeExt != 999.0) {
+    tailEdgeX -= (m_tailLength - m_tailEdgeExt);
+  }
+  Path edgeLimitSquare;
+  edgeLimitSquare.push_back(Point(noseEdgeX, -(m_noseWidth/2 + 1)));
+  edgeLimitSquare.push_back(Point(tailEdgeX, -(m_tailWidth/2 + 1)));
+  edgeLimitSquare.push_back(Point(tailEdgeX, (m_tailWidth/2 + 1)));
+  edgeLimitSquare.push_back(Point(noseEdgeX, (m_noseWidth/2 + 1)));
+  edgeLimitSquare.push_back(Point(noseEdgeX, -(m_noseWidth/2 + 1)));
+  
```

Comment:
I thought about this when I was making the initial changes but I think this should always work except if the board has negative side cut (not sure if this is an option)- i.e because of the sidecut the whole length of the ski/board between the hips will fit inside a trapezoid defined by the hips.

Actually I've just realised there is one situation that could cause a problem - if someone selects a negative nose or tail extension - not sure why they would want an edge shorter than the effective length but i guess it is a conceivable option... 

I'll switch to using a square as you suggest but i'll make it as wide as the widest of the nose, tail or **waist** - incase negative sidecut is possible (or this option is ever added).